projects/demo-playwright/tests/kit/input-date/input-date.spec.ts

```diff
@@ -0,0 +1,146 @@
+            await api.prepareBeforeScreenshot();
+            await expect(page).toHaveScreenshot('04-input-date.png');
+        });
+
+        test.describe('Invalid date cases', () => {
+            test('does not accept day > 31', async ({page}) => {
+                await tuiGoto(page, '/components/input-date/API');
+                await input.focus();
+                await page.keyboard.type('35');
+                await expect(input).toHaveJSProperty('selectionStart', 1);
+                await expect(input).toHaveJSProperty('selectionEnd', 1);
+                await expect(input).toHaveScreenshot('05-input-date.png');
+            });
+
+            test('does not accept month > 12', async ({page}) => {
+                await tuiGoto(page, '/components/input-date/API');
+                await input.focus();
+                await page.keyboard.type('1715');
+                await expect(input).toHaveJSProperty('selectionStart', 4);
+                await expect(input).toHaveJSProperty('selectionEnd', 4);
+                await expect(input).toHaveScreenshot('06-input-date.png');
+            });
+
+            test('Type 999999 => 09.09.9999', async ({page}) => {
+                await tuiGoto(page, '/components/input-date/API');
+                await input.focus();
+                await page.keyboard.type('999999');
+                await expect(input).toHaveJSProperty('selectionStart', 10);
+                await expect(input).toHaveJSProperty('selectionEnd', 10);
+                await expect(input).toHaveScreenshot('07-input-date.png');
+            });
+        });
+    });
+
+    test.describe('Mobile', () => {
+        test.use({
+            viewport: {width: 414, height: 896}, // Iphone XR
+            userAgent: TUI_PLAYWRIGHT_MOBILE_USER_AGENT,
+        });
+
+        ['russian', 'english', 'spanish'].forEach(language => {
+            test(`mobile calendar - ${language}`, async ({page}) => {
+                await tuiGoto(page, '/components/input-date', {language});
+                const api = new TuiDocumentationPagePO(page);
+                const example = api.getExample('#base');
+
+                await example.scrollIntoViewIfNeeded();
+                await api.prepareBeforeScreenshot(':not(#base)');
+                await example.locator('tui-input-date .t-icon tui-svg').click();
+                await expect(page).toHaveScreenshot(`08-input-date-${language}.png`);
+            });
+        });
+
+        test('native data picker', async ({page}) => {
+            await tuiGoto(page, '/components/input-date');
+            const example = new TuiDocumentationPagePO(page).getExample(
+                '#native-input-date',
+            );
+
+            await example.scrollIntoViewIfNeeded();
+            await example.locator('tui-input-date input[type="date"]').fill('2010-03-14');
+            await example
+                .locator('tui-input-date input[type="date"]')
+                .dispatchEvent('change');
```

Comment:
Probably it is cypress legacy. 
I've tried to remove this line and get the same snapshot.

Probably, Playwright already dispatches this action for `<input type="date" />`
https://playwright.dev/docs/input#text-input

Reply:
fixed